### PR TITLE
fix white scan error01

### PR DESCRIPTION
**Pull request by victorwangstone**
- fix linksdk_demo cannot use file system error
- fix whitescan error
- fix new white scan error

### CODE REVIEW

**Comment by skylarCai** [comment]
感谢提交 Pull Requests！
Thanks for submitting a pull request.

**Comment by skylarCai** [comment]
Security Check : 5265ef21672912e19fd1a4b63faf26ded52b0328

**Comment by skylarCai** [comment]
auto build success,number:150

**Review by wuxn9999 — APPROVED** [review]
https://github.com/alibaba/AliOS-Things/pull/1682
https://github.com/alibaba/AliOS-Things/pull/1693
和这两个重复了。本pr包含了两外两个pr的内容。请拆分开来。

File: components/linksdk/components/mqtt-upload/aiot_mqtt_upload_api.c [hunk 1]
```diff
@@ -1445,7 +1445,7 @@ int32_t aiot_mqtt_upload_send_data(void *handle, char *file_name,
 
 aiot_mqtt_upload_result_t aiot_mqtt_upload_process(void *handle) {
     mqtt_upload_handle_t *up_handle = (mqtt_upload_handle_t *)handle;
-    uint16_t file_name_len = 0;
+    uint16_t file_name_len = 0, upload_id_len = 0;
```
Comment (skylarCai) [inline components/linksdk/components/mqtt-upload/aiot_mqtt_upload_api.c:1448]: LGFM

**Review by skylarCai — COMMENTED** [review]
no security risk

**Pull request merged**